### DISCUSSION TITLE
It seems many to many model not generating

### DISCUSSION THREAD

**Discussion by aramis-it** [prisma/prisma#5167] [Q&A] [answered]
```prisma
model table1 {
  id           Int            @id @default(autoincrement())
  title        String?
  table1table2 table1table2[] @relation("table1table2Totable1")
}

model table2 {
  id           Int            @id @default(autoincrement())
  title        String?
  table1table2 table1table2[]
}

model table1table2 {
  table1Id Int
  table2Id Int
  table1   table1 @relation("table1table2Totable1", fields: [table1Id], references: [id])
  table2   table2 @relation(fields: [table2Id], references: [id])

  @@id([table1Id, table2Id])
  @@index([table1Id], name: "fk__Table1Table2_table11_idx")
  @@index([table2Id], name: "fk__Table1Table2_table21_idx")
  @@map("_table1table2")
}

```

**Comment by ryands17** [comment, marked as answer]
Hey @ulziibadrakh-p 👋 

Could you explain a bit about what issue are you facing?

**Reply by aramis-it** [reply]
I thought if M:N relation correctly generated after introspect, the model should look like below. Am I right?

```prisma
model table1 {
  id           Int            @id @default(autoincrement())
  title        String?
  table2s  table2[]   // expected
  table1table2 table1table2[] @relation("table1table2Totable1")  // instead this
}
```

**Reply by ryands17** [reply]
If you want a direct mapping to `table2`, you would need to model your table as defined [here](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#conventions-for-relation-tables-in-implicit-m-n-relations) as this is an [implicit many-to-many relationship](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#implicit-many-to-many-relations). So your SQL would look like:

```sql
create table "table1" (
	"id" serial primary key,
	"title" text
);

create table "table2" (
	"id" serial primary key,
	"title" text
);

create table "_table1Totable2" (
"A" integer not null references "table1"("id"),
"B" integer not null references "table2"("id")
);

create unique index "_table1Totable2_AB_unique" on "_table1Totable2"("A", "B");
create index "_table1Totable2_B_index" on "_table1Totable2"("B");
```

And after running `prisma introspect` you will get the following schema:

```prisma
model table1 {
  id     Int      @id @default(autoincrement())
  title  String?
  table2 table2[]
}

model table2 {
  id     Int      @id @default(autoincrement())
  title  String?
  table1 table1[]
}
```

**Reply by aramis-it** [reply]
Ok thank you it works.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
